src/PJ_robin.c

```diff
@@ -77,10 +77,10 @@ static const struct COEFS Y[] = {
 static XY s_forward (LP lp, PJ *P) {           /* Spheroidal, forward */
     XY xy = {0.0,0.0};
     int i;
-    double dphi;
+    double dphi = 0.0;
     (void) P;
 
-    i = (int)floor((dphi = fabs(lp.phi)) * C1);
+    i = lp.phi == lp.phi ? (int)floor((dphi = fabs(lp.phi)) * C1) : -1;
```

Comment:
Is this pattern commonly used and recognized? I find it quite unintuitive and will definitely not understand this the next time I stumble on this part of the code. 

Would it be possible to create a well-named function that takes care of this? Like `pj_avoid_nan_on_assignment()` (or something better...). Otherwise comments describing what's going on would be welcome.

Reply:
I would suggest adding an internal pj_is_nan() function to encapsulate NaN testing

Reply:
Works for me. I've got no idea how to write such a function, but I agree that it would be a better way to go in these situations.

Reply:
Just
```
int pj_is_nan(double x)
{
   /* cppcheck-suppress duplicateExpression */
   return x != x;
}
```

Reply:
Agreed.  `pj_is_nan` would be less confusing.

Reply:
Let's go with that then.

@schwehr can you add that function to this PR?

Reply:
I gave it a go with the simplest possible implementation in pj_internal.c.

I avoided messing with isnan or other GCC custom craziness as seen here:

https://trac.osgeo.org/gdal/browser/trunk/gdal/port/cpl_port.h?rev=41149#L575

And I didn't patch other instances of x == x nan checks.

Reply:
Good call. We can add craziness when needed.

> And I didn't patch other instances of x == x nan checks.

We can fix the rest as we come across them. Or if a master grepper want to search them out they can be done once and for all.